File: uxmobile/src/main/java/sk/brecka/uxmobile/model/event/CustomEvent.java
package sk.brecka.uxmobile.model.event;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.Map;

/**
 * Created by matej on 28.11.2017.
 */

public class CustomEvent extends Event {
    protected static final int INDEX_EVENT_NAME = 2;
    protected static final int INDEX_PAYLOAD = 3;

    private final String mEventName;
    private final Map<String, String> mPayload;


    public CustomEvent(long startTime, String eventName, Map<String, String> payload) {
        super(startTime);
        mEventName = eventName;
        mPayload = payload;
    }

    public CustomEvent(long startTime, String eventName) {
        this(startTime, eventName, null);
    }

    @Override
    protected String getType() {
        return TYPE_CUSTOM;
    }

    @Override
    public JSONArray toJson() throws JSONException {
        return super.toJson()
                .put(INDEX_EVENT_NAME, mEventName)
                // prazdny json ak nema payload
                .put(INDEX_PAYLOAD, mPayload != null ? new JSONObject(mPayload) : new JSONObject());
    }
}

File: uxmobile/src/main/java/sk/uxtweak/uxmobile/model/event/CustomEvent.kt
package sk.uxtweak.uxmobile.model.event

import org.json.JSONArray
import org.json.JSONObject

class CustomEvent(
    startTime: Long,
    private val eventName: String,
    private val payload: Map<String, String> = mapOf()
) : Event(startTime) {
    override fun getType() = TYPE_CUSTOM

    override fun toJson(): JSONArray = super.toJson()
        .put(INDEX_EVENT_NAME, eventName)
        .put(INDEX_PAYLOAD, JSONObject(payload))

    companion object {
        const val INDEX_EVENT_NAME = 2
        const val INDEX_PAYLOAD = 3
    }
}
